processing/mars_stealer/mars_stealer.py

```diff
@@ -0,0 +1,336 @@
+from fame.core.module import ProcessingModule
+from fame.common.exceptions import ModuleInitializationError
+
+try:
+    import re
+    import json
+    import base64
+    import r2pipe
+    from typing import Any, List
+    from collections import namedtuple
+    from cryptography.hazmat.primitives.ciphers import Cipher, algorithms
+
+    Section = namedtuple("Section",
+                         ["name", "id", "paddr", "size",
+                          "vaddr", "vsize", "perm"])
+    HAVE_CSPARSER = True
+except ImportError as err:
+    print(err)
+    HAVE_CSPARSER = False
+
+
+def unxor(string: List[Any], key: List[Any]) -> str:
+    """Method to unxor obfuscated data from llcppc section"""
+
+    unxored = ""
+
+    for c1, c2 in zip(string, key):
+        unxored += chr(c1 ^ c2)
+
+    return unxored
+
+
+def decrypt_rc4(key: bytes, ciphertext: bytes) -> bytes:
+    """Decrypt RC4 encrypt data"""
+
+    algorithm = algorithms.ARC4(key)
+    cipher = Cipher(algorithm, mode=None)
+    decryptor = cipher.decryptor()
+    cleartext = decryptor.update(ciphertext)
+
+    return cleartext
+
+
+class MarsStealerC2Extractor(ProcessingModule):
+
+    name: str = "mars_stealer_c2"
+    description: str = "Mars Stealer command and control configuration extractor"
+
+    def initialize(self):
+        if not HAVE_CSPARSER:
+            raise ModuleInitializationError(self, "Missing dependency")
+
+    def each(self, target):
+        r2 = MarsStealerC2Extractor.analyze_file(target)
+        sections = MarsStealerC2Extractor.map_sections(r2)
+        try:
+            c2 = self.get_c2(r2, sections)
+        except Exception as err:
+            print(f"Error getting c2: {err}")
+
+        if c2 is None:
+            return False
+
+        c2 = list(c2)
+        if 'http' not in c2[0]:
+            dest = c2[0]
+            if not c2[1].startswith("/"):
+                c2[1] = f"/{c2[1]}"
+            c2 = ''.join(c2)
+            self.add_ioc(dest, ["marsstealer", "c2"])
+            if not c2.startswith('http://') and not c2.startswith('https://'):
+                for schema in ["http://", "https://"]:
+                    self.add_ioc(f"{schema}{c2}", ["marsstealer", "c2"])
+        else:
+            dest = c2[1]
+            if not c2[2].startswith("/"):
+                c2[2] = f"/{c2[2]}"
+            c2_url = ''.join(c2)
+            self.add_ioc(dest, ["marsstealer", "c2"])
+            self.add_ioc(c2_url, ["marsstealer", "c2"])
+
+        self.add_tag("marsstealer")
```

Comment:
As mentioned earlier, this is done automatically if this is the module's name. You should probably replace it by `self.add_probable_name("Mars Stealer")`